### DISCUSSION TITLE
Linker errors while building imgui examples

### DISCUSSION THREAD

**Discussion by starbird1975** [ocornut/imgui#5950] [New Users Build/Link/Run issues ONLY!]
Hello all

While trying to build the example 'example_sdl_sdlrenderer' using Visual Studio Community 22 and
using the provided solution file (imgui_examples.sln) I got a bunch of linker errors. They were easily
fixed by adding winmm.lib, setupapi.lib and version.lib to Configuration Properties/Linker/Input/Additional Dependencies.
I guess it's the same for the other examples as well. I wonder whether other people have encountered this as well,
or is this maybe a Visual Studio Configuration issue? In any case I just wanted to report this in case other people
stumble over this problem.

**Comment by PathogenDavid** [comment]
You shouldn't have to do anything besides provide a path to your SDL installation via `SDL2_DIR` so you've likely done something odd.

**Comment by starbird1975** [comment]
I built SDL (static libs) from source with cmake from the latest release branch and made sure the example finds the SDL includes and the libs SDL.lib/SDLmain.lib by editing the Properties. It was all straightforward, except for having to add those three windows related libraries
by hand. Note that the unresolved externals were all referenced by SDL functions.

**Reply by PathogenDavid** [reply]
> SDL (static libs)

Yeah, that would do it. The examples are assuming SDL was built as a DLL since that's how the SDL team distributes it.

When you build things as static libraries you become responsible for linking their dependencies when you consume them.
